src/uu/vmstat/src/vmstat.rs

```diff
@@ -4,187 +4,122 @@
 // file that was distributed with this source code.
 
 mod parser;
+mod picker;
 
-use clap::{crate_version, Command};
+#[cfg(target_os = "linux")]
+use crate::picker::{get_pickers, Picker};
+use clap::value_parser;
+#[allow(unused_imports)]
+use clap::{arg, crate_version, ArgMatches, Command};
 #[allow(unused_imports)]
 pub use parser::*;
-#[cfg(target_os = "linux")]
-use procfs::{Current, CurrentSI};
-use uucore::error::UResult;
+#[allow(unused_imports)]
+use uucore::error::{UResult, USimpleError};
 use uucore::{format_usage, help_about, help_usage};
 
 const ABOUT: &str = help_about!("vmstat.md");
 const USAGE: &str = help_usage!("vmstat.md");
 
 #[uucore::main]
 pub fn uumain(args: impl uucore::Args) -> UResult<()> {
-    let _matches = uu_app().try_get_matches_from(args)?;
-
-    let mut section: Vec<String> = vec![];
-    let mut title: Vec<String> = vec![];
-    let mut data: Vec<String> = vec![];
-    let mut data_len_excess = 0;
-
+    #[allow(unused)]
+    let matches = uu_app().try_get_matches_from(args)?;
     #[cfg(target_os = "linux")]
-    let func = [
-        concat_helper(get_process_info),
-        concat_helper(get_memory_info),
-        concat_helper(get_swap_info),
-        concat_helper(get_io_info),
-        concat_helper(get_system_info),
-        concat_helper(get_cpu_info),
-    ];
-
-    #[cfg(not(target_os = "linux"))]
-    let func: [ConcatFunc; 0] = [];
-
-    func.iter()
-        .for_each(|f| f(&mut section, &mut title, &mut data, &mut data_len_excess));
-
-    println!("{}", section.join(" "));
-    println!("{}", title.join(" "));
-    println!("{}", data.join(" "));
+    {
+        // validate unit
+        if let Some(unit) = matches.get_one::<String>("unit") {
+            if !["k", "K", "m", "M"].contains(&unit.as_str()) {
+                Err(USimpleError::new(
+                    1,
+                    "-S requires k, K, m or M (default is KiB)",
+                ))?;
+            }
+        }
+
+        let one_header = matches.get_flag("one-header");
+        let no_first = matches.get_flag("no-first");
+
+        let delay = matches.get_one::<u64>("delay");
+        let count = matches.get_one::<u64>("count");
+        let mut count = if let Some(count) = count {
+            if *count == 0 {
+                Some(1)
+            } else {
+                Some(*count)
+            }
+        } else {
+            None
+        };
```

Comment:
I think you can simplify this `if/let` with `map`:
```suggestion
        let mut count = count.map(|c| if *c == 0 { 1 } else { *c });
```